mkosi/config.py

```diff
@@ -2068,6 +2076,37 @@ def parse_config(argv: Sequence[str] = ()) -> tuple[MkosiArgs, tuple[MkosiConfig
     # Compare inodes instead of paths so we can't get tricked by bind mounts and such.
     parsed_includes: set[tuple[int, int]] = set()
 
+    def expand_specifiers(text: str, namespace: argparse.Namespace, defaults: argparse.Namespace) -> str:
+        percent = False
+        result = ""
```

Comment:
This will have quadratic runtime, since a new string will be allocated every time a new character is added. if you make this a list an `"".join(result)` at the end it will still bump the list around in memory a bit when a new character is appended, but the allocations will average out and the runtime should be closer to linear in the length of `text`.